chain/network/src/peer_manager/peer_store.rs

```diff
@@ -85,23 +85,25 @@ impl PeerStore {
             let mut peer_state: KnownPeerState = value.try_into()?;
             // Mark loaded node last seen to now, to avoid deleting them as soon as they are loaded.
             peer_state.last_seen = to_timestamp(Utc::now());
-            match peer_state.status {
-                KnownPeerStatus::Banned(_, _) => {}
-                _ => peer_state.status = KnownPeerStatus::NotConnected,
+            peer_state.status = match peer_state.status {
+                banned_status @ KnownPeerStatus::Banned(_, _) => banned_status,
+                _ => KnownPeerStatus::NotConnected,
             };
 
-            if let Some(current_peer_state) = peer_states.get_mut(&peer_id) {
-                // This peer is a boot node and was already added so skip.
-                if peer_state.status.is_banned() {
-                    current_peer_state.status = peer_state.status;
+            match peer_states.entry(peer_id) {
+                Entry::Occupied(mut current_peer_state) => {
+                    // This peer is a boot node and was already added so skip.
+                    if peer_state.status.is_banned() {
+                        current_peer_state.get_mut().status = peer_state.status;
```

Comment:
comment about this case (I guess this means that peer was banned in the past, right?)